task_sdk/src/airflow/sdk/definitions/dag.py

```diff
@@ -0,0 +1,1048 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import copy
+import functools
+import itertools
+import logging
+import os
+import pathlib
+import sys
+import weakref
+from collections import abc
+from collections.abc import Collection, Iterable, Iterator
+from datetime import datetime, timedelta, timezone
+from inspect import signature
+from re import Pattern
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    ClassVar,
+    Self,
+    Union,
+    cast,
+)
+from urllib.parse import urlsplit
+
+import attrs
+import jinja2
+import re2
+from dateutil.relativedelta import relativedelta
+
+import airflow.templates
+from airflow import settings
+from airflow.assets import Asset, AssetAlias, BaseAsset
+from airflow.configuration import conf as airflow_conf
+from airflow.exceptions import (
+    DuplicateTaskIdFound,
+    FailStopDagInvalidTriggerRule,
+    ParamValidationError,
+    TaskNotFound,
+)
+from airflow.models.param import DagParam
+from airflow.sdk.definitions.abstractoperator import AbstractOperator
+from airflow.sdk.definitions.baseoperator import BaseOperator
+from airflow.sdk.types import NOTSET
+from airflow.stats import Stats
+from airflow.timetables.base import Timetable
+from airflow.timetables.interval import CronDataIntervalTimetable, DeltaDataIntervalTimetable
+from airflow.timetables.simple import (
+    AssetTriggeredTimetable,
+    ContinuousTimetable,
+    NullTimetable,
+    OnceTimetable,
```

Comment:
Where should Timetables live?

User can pass it to the DAG --- so maybe SDK.

Yeah all of it is intertwined


[21:27](https://astronomer.slack.com/archives/DJ28R0NKZ/p1729110420516149)
User can pass it to the DAG --- so maybe SDK